netbox/utilities/api.py

```diff
@@ -98,20 +105,24 @@ def get_prefetches_for_serializer(serializer_class, fields_to_include=None):
         # If the serializer field does not map to a discrete model field, skip it.
         try:
             field = model._meta.get_field(model_field_name)
-            if isinstance(field, (RelatedField, ManyToOneRel, GenericForeignKey)):
-                prefetch_fields.append(field.name)
         except FieldDoesNotExist:
             continue
 
+        # Prefetch for tags
```

Comment:
This code is close but I think this is way too specific checking for a field named 'tags'. There shouldn't be anything special about tags - we should be able to check generically for M2M or FK field here.

Reply:
If this is needed can you do something like:
```
if isinstance(field, models.ManyToManyField):
    prefetch_fields.append(field.name)
elif serializer_field and source_field is None:
   ...
```